internal/middleware/paginate.go

```diff
@@ -0,0 +1,37 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/rs/zerolog"
+)
+
+type ctxKeyType int
+
+const (
+	OffsetCtxKey ctxKeyType = iota
+	LimitCtxKey
+)
```

Comment:
The general practice is not to export context keys, make them unexported and provide functions `Offset/WithOffset` and `Limit/WithLimit` that will lookup/store value in the context. This is a nice encapsulation and you also make sure that nobody will mess up with the value in the context. Also the functions will do the conversion to the proper type (int in this case).

Finally, I think it always makes sense to have both values and not one, so in this case create a new type that will have both ints and work with this type instead of individually. This struct could also have a boolean indicating if there is actually pagination or not - this will depend on the endpoint.

```
type Pagination struct {
  Active bool
  Limit int
  Offset int
}
```

The middleware could always insert either pagination that is not active (basically just empty value: `Pagination{}` = not active) or an active one with proper numbers. Then you can always be sure you never get a `nil` out of context and the code will be pretty:

```
pg := middleware.Pagination(ctx)
if pg.Active {
  limit := pg.Limit
  offset := pg.Offset
  // ...
}
```